src/plzinit/plugins.go

```diff
@@ -0,0 +1,286 @@
+}
+
+func writeCCConfigFields(file ast.File) ast.File {
+	configMap := map[string]string{
+		"cctool":             "CCTool",
+		"cpptool":            "CPPTool",
+		"ldtool":             "LDTool",
+		"artool":             "ARTool",
+		"defaultoptcflags":   "DefaultOptCFlags",
+		"defaultdbgcflags":   "DefaultDbgCFlags",
+		"defaultoptcppflags": "DefaultOptCppFlags",
+		"defaultdbgcppflags": "DefaultDbgCppFlags",
+		"defaultldflags":     "DefaultLdFlags",
+		"pkgconfigpath":      "PkgConfigPath",
+		"testmain":           "TestMain",
+		"dsymtool":           "DsymTool",
+	}
+
+	return writeFieldsToConfig("cc", file, configMap)
+}
+
+func writeJavaConfigFields(file ast.File) ast.File {
+	configMap := map[string]string{
+		"javactool":          "JavacTool",
+		"javacworker":        "JavacWorker",
+		"junitrunner":        "JunitRunner",
+		"defaulttestpackage": "DefaultTestPackage",
+		"releaselevel":       "ReleaseLevel",
+		"targetlevel":        "TargetLevel",
+		"javacflags":         "JavacFlags",
+		"javactestflags":     "JavacTestFlags",
+		"defaultmavenrepo":   "MavenRepo",
+		"toolchain":          "Toolchain",
+	}
+
+	return writeFieldsToConfig("java", file, configMap)
+}
+
+func writeFieldsToConfig(plugin string, file ast.File, configMap map[string]string) ast.File {
+	section := "Plugin"
+
+	// Check for existing plugin section
+	if s := file.MaybeGetSection(section, plugin); s != nil {
+		log.Warningf("Plugin config section already exists, so init did nothing.")
+		return file
+	}
+
+	// Write plugin target value
+	file = ast.InjectField(file, "Target", "//plugins:"+plugin, section, plugin, false)
+
+	// Migrate any existing language fields to their plugin equivalents
+	foundSection := false
+	for _, s := range file.Sections {
+		if s.Key == plugin {
+			foundSection = true
+			for _, field := range s.Fields {
+				if plugVal, ok := configMap[strings.ToLower(field.Name)]; ok {
+					file = ast.InjectField(file, plugVal, field.Value, section, plugin, false)
+				}
+			}
+		}
+	}
+
+	// If we found nothing, add a section with default values
+	if !foundSection {
+		config, err := core.ReadDefaultConfigFiles(nil)
+		if err != nil {
+			log.Fatalf("Error reading config file: %s", err)
+		}
+
+		// Build plugin target so we can pull the default values
+		state := core.NewBuildState(config)
+		buildLabel := core.NewBuildLabel("plugins", plugin)
+		plz.Run([]core.BuildLabel{buildLabel}, nil, state, config, state.TargetArch)
+		subrepo := state.Graph.Subrepo(plugin)
+		if subrepo == nil {
+			log.Fatalf("Failed to get subrepo %v", plugin)
+		}
+		if err = subrepo.LoadSubrepoConfig(); err != nil {
+			panic(err)
+		}
+		config = subrepo.State.Config
+		for _, v := range config.PluginConfig {
+			if len(v.DefaultValue) == 0 {
+				file = ast.InjectField(file, v.ConfigKey, "", section, plugin, v.Repeatable)
+			} else {
+				// Check for build labels and resolve relative to host
+				val := v.DefaultValue[0]
```

Comment:
If `v.DefaultValue` is an arrary, it might be better to iterate through all the values as `v.DefaultValue[1]` could also be a build label that needs normalisation?